src/main/java/com/faforever/client/teammatchmaking/PartyMemberItemController.java

```diff
@@ -0,0 +1,176 @@
+package com.faforever.client.teammatchmaking;
+
+import com.faforever.client.chat.ChatChannelUser;
+import com.faforever.client.chat.ChatService;
+import com.faforever.client.chat.ChatUserContextMenuController;
+import com.faforever.client.chat.CountryFlagService;
+import com.faforever.client.chat.avatar.AvatarService;
+import com.faforever.client.fx.Controller;
+import com.faforever.client.game.Faction;
+import com.faforever.client.game.PlayerStatus;
+import com.faforever.client.i18n.I18n;
+import com.faforever.client.player.Player;
+import com.faforever.client.player.PlayerService;
+import com.faforever.client.teammatchmaking.Party.PartyMember;
+import com.faforever.client.theme.UiService;
+import com.faforever.client.util.RatingUtil;
+import com.google.common.base.Strings;
+import javafx.application.Platform;
+import javafx.beans.binding.BooleanBinding;
+import javafx.css.PseudoClass;
+import javafx.event.ActionEvent;
+import javafx.scene.Node;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.image.ImageView;
+import javafx.scene.input.ContextMenuEvent;
+import javafx.scene.layout.HBox;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+import java.lang.ref.WeakReference;
+
+import static javafx.beans.binding.Bindings.createObjectBinding;
+import static javafx.beans.binding.Bindings.createStringBinding;
+
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class PartyMemberItemController implements Controller<Node> {
+
+  private static final PseudoClass LEADER_PSEUDO_CLASS = PseudoClass.getPseudoClass("leader");
+  private static final PseudoClass PLAYING_PSEUDO_CLASS = PseudoClass.getPseudoClass("playing");
+
+  private final CountryFlagService countryFlagService;
+  private final AvatarService avatarService;
+  private final PlayerService playerService;
+  private final TeamMatchmakingService teamMatchmakingService;
+  private final UiService uiService;
+  private final ChatService chatService;
+  private final I18n i18n;
+
+  public Node playerItemRoot;
+  public ImageView avatarImageView;
+  public ImageView countryImageView;
+  public ImageView leagueImageView;
+  public Button kickPlayerButton;
+  public Label clanLabel;
+  public Label usernameLabel;
+  public Label leagueLabel;
+  public Label gameCountLabel;
+  public Label uefLabel;
+  public Label cybranLabel;
+  public Label aeonLabel;
+  public Label seraphimLabel;
+  public Label crownLabel;
+  public HBox playerCard;
+  public ImageView playerStatusImageView;
+
+  private Player player;
+  //TODO: this is a bit hacky
```

Comment:
Don't tell me it's hacky without telling what and why. I assume the WeakReference is meant here, but it I don't know for sure.

Reply:
I don't know either. That comment is from Geosearchef.

Reply:
a214cbdf

Reply:
The contextmenu is separable from the chatUser which might be the thing to do here as I don't understand why the chatUser is necessary if all we want is a context menu

Reply:
This also seems like it would be a different context menu